### PR TITLE
Set Event.http.request.url on a best effort basis - 2

### PR DESCRIPTION
**Pull request by rish691**
Option 2 of https://github.com/hypertrace/hypertrace-ingester/pull/99

The changes include: 
1. Update HttpFieldsGenerator#maybeSetHttpUrlForOtelFormat to overwrite relative url
2. Update the logic to fetch http url in SpanEventViewGenerator

### CODE REVIEW

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java [hunk 1]
```diff
@@ -569,13 +569,27 @@ private static URL getNormalizedUrl(String url) throws MalformedURLException {
...
     }
     Optional<String> url = HttpSemanticConventionUtils.getHttpUrlForOTelFormat(attributeValueMap);
     url.ifPresent(requestBuilder::setUrl);
   }
+
+  static boolean isAbsoluteUrl(String urlStr) {
+    try {
+      URL url = getNormalizedUrl(urlStr);
+      return url.toString().equals(urlStr);
+    } catch (MalformedURLException e) {
```
Comment (tim-mwangi) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java:590]: Can we add unit tests for these negative scenarios?(If possible)

**Comment by buchi-busireddy** [comment]
@rish691 So, if I understand it right, you're moving to a model where:
- `span.http.request.url` will be set only with the full URL
- if the full URL is missing, the consumer can decide what to do. View-gen is using path because it's for display in the UI.

**Comment by rish691** [comment]
> @rish691 So, if I understand it right, you're moving to a model where:
> 
> * `span.http.request.url` will be set only with the full URL
> * if the full URL is missing, the consumer can decide what to do. View-gen is using path because it's for display in the UI.

So, presently also `span.http.request.url` is set in case of full URL. The change in this PR is to just the increase the chances of it being set to a full url. 

If we still see problems downstream, we can revisit the logic of setting `span.http.request.url` as null.

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java (hunk 1 above)
Comment (rish691) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java:590]: The java implementation for URL conversion seems to be able to parse all the strings! I will update the test, when I find one.

**Review by buchi-busireddy — APPROVED** [review]
